maplibre-navigation-core/src/commonMain/kotlin/org/maplibre/navigation/core/navigation/MapLibreNavigation.kt

```diff
@@ -90,6 +90,29 @@ open class MapLibreNavigation @JvmOverloads constructor(
     var fasterRouteEngine: FasterRoute = FasterRouteDetector(options),
     val routeUtils: RouteUtils = RouteUtils(),
 ) {
+    /**
+     * Convenience constructor for tests. Enables injection of navigation engine mock.
+     */
+    internal constructor(
+        options: MapLibreNavigationOptions = MapLibreNavigationOptions(),
+        locationEngine: LocationEngine,
+        cameraEngine: Camera = SimpleCamera(),
+        snapEngine: Snap = SnapToRoute(),
+        offRouteEngine: OffRoute = OffRouteDetector(),
+        fasterRouteEngine: FasterRoute = FasterRouteDetector(options),
+        routeUtils: RouteUtils = RouteUtils(),
+        mapLibreNavigationEngine: MapLibreNavigationEngine
+    ) : this(
+        options,
+        locationEngine,
+        cameraEngine,
+        snapEngine,
+        offRouteEngine,
+        fasterRouteEngine,
+        routeUtils,
+    ) {
+        this.mapLibreNavigationEngine = mapLibreNavigationEngine
```

Comment:
We should thinking about adding this to the main constructor. And allowing also to set an own navigation engine. But this should be a topic for an other PR.